include/alpaka/core/RuntimeMacros.hpp

```diff
@@ -0,0 +1,48 @@
+/* Copyright 2022  Andrea Bocci, Mehmet Yusufoglu, René Widera, Aurora Perego
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+// Implementation details
+#include "alpaka/core/Sycl.hpp"
+
+//! ALPAKA_THROW_ACC either aborts(terminating the program and creating a core dump) or throws std::runtime_error
+//! depending on the Acc. The std::runtime_error exception can be catched in the catch block.
+//!
+//! CUDA __trap function triggers std::runtime_error but can be catched during wait not exec.
+//!
+//! The OpenMP specification mandates that exceptions thrown by some thread must be handled by the same thread.
+//! Therefore std::runtime_error thrown by ALPAKA_THROW_ACC aborts the the program for OpenMP backends. If needed
```

Comment:
please add HIP and SYCL behaviour too

Reply:
this is a oneAPI implementation details, but you could try calling 
```c++
#ifdef __GLIBC__
extern "C" {
extern __DPCPP_SYCL_EXTERNAL void __assert_fail(const char *expr,
                                                const char *file,
                                                unsigned int line,
                                                const char *func);
}
#elif defined(_WIN32)
extern "C" {
// TODO: documented C runtime library APIs must be recognized as
//       builtins by FE. This includes _dpcomp, _dsign, _dtest,
//       _fdpcomp, _fdsign, _fdtest, _hypotf, _wassert.
//       APIs used by STL, such as _Cosh, are undocumented, even though
//       they are open-sourced. Recognizing them as builtins is not
//       straightforward currently.
extern __DPCPP_SYCL_EXTERNAL void _wassert(const wchar_t *wexpr,
                                           const wchar_t *wfile, unsigned line);
}
#endif
```

Reply:
Added. For sycl it turned out to be my tests and @AuroraPerego and @fwyzard's tests both generated a runtime error saying "...SYCL exception: Unresolved Symbol <abort> " etc. Therefore assert(false) is used and related comments are added.